Vulkan crashes when `EnableModel=0`

### Describe the bug

Vulkan `RenderMode` works fine when `EnableModel=1`, otherwise Ikemen crashes with the following stacktrace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9df325]

goroutine 1 [running, locked to thread]:
main.(*Renderer_VK).BeginFrame(0xc00032a000, 0x0)
        github.com/ikemen-engine/Ikemen-GO/src/render_vk.go:5003 +0x9a5
main.(*System).init(0x1160bc0, 0x18?, 0x0?)
        github.com/ikemen-engine/Ikemen-GO/src/system.go:415 +0xafb
main.realMain()
        github.com/ikemen-engine/Ikemen-GO/src/main.go:184 +0x414
main.main()
        github.com/ikemen-engine/Ikemen-GO/src/main.go:57 +0x17
```

### To Reproduce

Run Ikemen with `RenderMode=Vulkan 1.3` and `EnableModel=0` and watch it crash.

### Expected behavior

No crash.

### Screenshots / Video

_No response_

### Engine Version (or source code date)

2026-01-05

### Operating system

Linux

### Extra context or search terms

_No response_

--- a/src/render_vk.go
+++ b/src/render_vk.go
@@ -181,6 +181,9 @@ func (r *Renderer_VK) newDummyCubeMapTexture() Texture {
 }
 
 func (t *Texture_VK) SetData(textureData []byte) {
+	if textureData == nil {
+		return
+	}
 	t.SetSubData(textureData, t.offset[0], t.offset[1], t.width, t.height)
 }
 func (t *Texture_VK) SetSubData(textureData []byte, x, y, width, height int32) {
@@ -5000,8 +5003,10 @@ func (r *Renderer_VK) BeginFrame(clearColor bool) {
 	r.currentSpriteTexture.palTexture = r.dummyTexture
 	r.VKState.spriteTexture = r.dummyTexture
 	r.VKState.palTexture = r.dummyTexture
-	r.modelProgram.uniformOffsetMap = make(map[interface{}]uint32)
-	r.modelProgram.uniformBufferOffset = 0
+	if r.modelProgram != nil {
+		r.modelProgram.uniformOffsetMap = make(map[interface{}]uint32)
+		r.modelProgram.uniformBufferOffset = 0
+	}
 }
 
 func (r *Renderer_VK) BlendReset() {

